control_panel_api/pagination.py

```diff
@@ -1,5 +1,37 @@
-from rest_framework.pagination import PageNumberPagination
+from rest_framework.pagination import PageNumberPagination, _positive_int
 
 
 class CustomPageNumberPagination(PageNumberPagination):
+    """This Pagination class allows a request to pass the page_size query param
+    for usual pagination functionality but adds custom whereby a value of
+    page_size=0 will return all items
+    """
     page_size_query_param = 'page_size'
+
+    def paginate_queryset(self, queryset, request, view=None):
+        """Override to check for page_size=0 if so then return all items else
+        execute original parent functionality
+        """
+        page_size = self.get_page_size(request)
+        if page_size != 0:
+            return super().paginate_queryset(queryset, request, view)
+
+        paginator = self.django_paginator_class(queryset, queryset.count())
+        self.page = paginator.page(1)
+
+        return list(self.page)
+
+    def get_page_size(self, request):
```

Comment:
This is all copied from `rest_framework.pagination.PageNumberPagination` right? So you can just not implement this and use the implementation in the parent class.

Reply:
Crucially I changed the boolean, that is all :/

Reply:
Stupid question, why comment this method make the test fail? Shouldn't it the use `get_page_size()` in the parent class which is same as this?

I'm probably missing something.

Reply:
It may have something to do with `_positive_int()` used internally, I'm having a quick look.

Reply:
Right, so it has something to do with the internal use of `_positive_int()` then.

Reply:
👍

Reply:
I agree with @xoen's original comment - use the parent class implementation, eg:
```python
def get_page_size(self, request):
    try:
        page_size = super().get_page_size(request)

    except ValueError:
        try:
            if int(request.query_params[self.page_size_query_param]) == 0:
                return 0
        except (KeyError, ValueError):
            pass

    return self.page_size
```

This way, you don't have to depend on a private function.

Reply:
I can't rely on a thrown exception as the parent hides it and always returns a default.